src/createServer.js

```diff
@@ -1,3 +1,55 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+const createServer = () => (http.createServer((req, res) => {
+  res.setHeader('Content-Type', 'application/json');
+
+  const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+  const toCase = Object.fromEntries(normalizedUrl.searchParams.entries())
+    .toCase;
+  const path = normalizedUrl.pathname.slice(1);
+  const errors = [];
+
+  if (!path) {
+    addError(errors, 'Text to convert is required. Correct request'
+      + ' is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".');
+  }
+
+  if (toCase
+      && !['SNAKE', 'KEBAB', 'UPPER', 'CAMEL', 'PASCAL'].includes(toCase)) {
+    addError(errors, 'This case is not supported.'
+      + ' Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.');
+  }
+
+  if (!toCase) {
+    addError(errors, '"toCase" query param is required.'
+      + ' Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".');
+  }
```

Comment:
move it to the separate function
```suggestion
  const errors = ...(...);
```